internalsdk/android_test.go

```diff
@@ -77,30 +78,48 @@ func (c testSession) IsPlayVersion() (bool, error)                       { retur
 func (c testSession) SetShowGoogleAds(enabled bool)                      {}
 func (c testSession) SetHasConfigFetched(enabled bool)                   {}
 func (c testSession) SetHasProxyFetched(enabled bool)                    {}
-func (c testSession) SetOnSuccess(enabled bool)                          {}
 func (c testSession) ChatEnable() bool                                   { return false }
 
+func (c testSession) SetOnSuccess(enabled bool) {
+	if !enabled {
+		return
+	}
+	// we need to wait for flashlight.fastConnect dialer to "test" and get a successful connection
+	// before any dialer can be used ("no top dialer" error)
+	select {
+	case c.rdy <- true:
+	default:
+	}
```

Comment:
I'm a bit confused by this code given the comment.  This code doesn't wait at all.  Would you mind explaining?

Reply:
Added more to the comment to explain that this is only part of the solution and we would still need to actually listen for `c.rdy`.